python_modules/dagster/dagster/_utils/schedules.py

```diff
@@ -1,18 +1,28 @@
 import datetime
+import functools
 from typing import Iterator, Optional, Sequence, Union
 
 import pendulum
 import pytz
-from croniter import croniter
+from croniter import croniter as _croniter
 
 import dagster._check as check
 from dagster._seven.compat.pendulum import to_timezone
 
 
+class CroniterShim(_croniter):
+    """Lightweight shim to enable caching certain values that may be calculated many times."""
+
+    @classmethod
+    @functools.lru_cache(maxsize=128)
```

Comment:
128 appears to be the default - how scientifically was that chosen here / should we be tempted at all to make this None so that it caches everything? The space of all possible cron strings seems pretty low in practice. If we're unlikely to ever hit past 128 in a single session then it probably does not matter a ton

Reply:
zero science -- was just a combination of it being the default, seeming much larger than the number of cron strings I'd expect to see in a single session, and unbounded caching felt scary